packages/svg-serializer/index.js

```diff
@@ -1,66 +1,134 @@
+  }
 
   var body = ['svg',
     {
-      width: width + 'mm',
-      height: height + 'mm',
+      width: width + options.unit,
+      height: height + options.unit,
       viewBox: ('0 0 ' + width + ' ' + height),
       version: '1.1',
       baseProfile: 'tiny',
       xmlns: 'http://www.w3.org/2000/svg',
       'xmlns:xlink': 'http://www.w3.org/1999/xlink'
     },
-    PathsToSvg(paths, bounds)
   ]
+  if (bounds) {
+    body = body.concat(convertObjects(objects, bounds, options))
+  }
 
   var svg = `<?xml version="1.0" encoding="UTF-8"?>
 <!-- Generated by OpenJSCAD.org -->
 <!DOCTYPE svg PUBLIC "-//W3C//DTD SVG 1.1 Tiny//EN" "http://www.w3.org/Graphics/SVG/1.1/DTD/svg11-tiny.dtd">
 ${stringify(body)}`
 
-  options && options.statusCallback && options.statusCallback({progress: 100})
+  options.statusCallback && options.statusCallback({progress: 100})
   return [svg]
 }
 
-const PathsToSvg = function (paths, bounds, options) {
-  // calculate offsets in order to create paths orientated from the 0,0 axis
+const getBounds = (objects) => {
+  let bounds = null
+  objects.forEach(function (object, i) {
+    if (isCAG(object) && object.sides.length > 0) {
+      let cagBounds = object.getBounds()
+      if (bounds !== null) {
+        bounds[0] = bounds[0].min(cagBounds[0])
+        bounds[1] = bounds[1].max(cagBounds[1])
+      } else {
+        bounds = cagBounds
+      }
+    }
+  })
+  return bounds
+}
+
+const convertObjects = (objects, bounds, options) => {
   var xoffset = 0 - bounds[0].x
   var yoffset = 0 - bounds[0].y
 
+  let contents = []
+  objects.forEach(function (object, i) {
+    options.statusCallback && options.statusCallback({progress: 100 * i / objects.length})
+    if (isCAG(object) && object.sides.length > 0) {
+      contents.push(convertCAG(object, [xoffset, yoffset], options))
+    }
+  })
+  return contents
+}
+
+const convertCAG = (object, offsets, options) => {
+  var paths = object.getOutlinePaths()
+  return convertPaths(paths, offsets, options)
+}
+
+const convertPaths = (paths, offsets, options) => {
   return paths.reduce(function (res, path, i) {
-    options && options.statusCallback && options.statusCallback({progress: 100 * i / paths.length})
-    return res.concat([['path', {d: dPath(path, xoffset, yoffset)}]])
+    return res.concat([['path', {d: convertPath(path, offsets, options)}]])
   }, ['g'])
 }
 
-const dPath = function (path, xoffset, yoffset) {
+const convertPath = (path, offsets, options) => {
   var pointindex
   var str = ''
   var numpointsClosed = path.points.length + (path.closed ? 1 : 0)
   for (pointindex = 0; pointindex < numpointsClosed; pointindex++) {
     var pointindexwrapped = pointindex
     if (pointindexwrapped >= path.points.length) pointindexwrapped -= path.points.length
     var point = path.points[pointindexwrapped]
+    let x = Math.round((point.x + offsets[0]) * options.decimals) / options.decimals
```

Comment:
I did not follow where these roundings come from ?

Reply:
Not sure. I thing @drom added the width and size calculations.

I was wondering the same about rounding values. At this point, the serializers are inconsistent when converting float numbers to text. Some round, some don’t.

@danmarshall I remember some changes to your library for rounding.  Is it necessary?

Reply:
@z3dev - the SVG rounding in Maker.js is more of a feature than a necessity. In case someone wants to export with just a few decimal places instead of what JavaScript typically spits out.

Reply:
@danmarshall thanks
@kaosat-dev the rounding of numbers was originally in the code. maybe @drom added those when converting to use the omnl library. we should be consistent if we decide to go down this road. should i remove the rounding?